zkevm-circuits/src/rlp_circuit_fsm.rs

```diff
@@ -572,13 +576,30 @@ impl<F: Field> RlpCircuitConfig<F> {
                 },
             );
 
+            // For the very last non-padding RLP instance, we have
+            //   tx_id' != tx_id && tx_id' == 0
+            cb.condition(
+                and::expr([
+                    is_padding_in_dt.expr(Rotation::next())(meta),
+                    not::expr(tx_id_check_in_dt.is_equal_expression.expr()),
+                ]),
+                |cb| {
+                    // byte_rev_idx == 1
+                    cb.require_equal(
+                        "byte_rev_idx is 1 at the last index",
+                        meta.query_advice(data_table.byte_rev_idx, Rotation::cur()),
+                        1.expr(),
+                    );
+                },
+            );
+
             cb.gate(meta.query_fixed(q_enabled, Rotation::cur()))
         });
 
         meta.lookup_any("byte value check", |meta| {
             let cond = and::expr([
                 meta.query_fixed(q_enabled, Rotation::cur()),
-                is_padding_in_dt.expr(Rotation::cur())(meta),
+                not::expr(is_padding_in_dt.expr(Rotation::cur())(meta)),
```

Comment:
We would suggest adding some negative tests to catch issues like this when changes are made to the circuit in the future.